geoips/plugins/modules/output_formatters/unprojected_image.py

```diff
@@ -60,22 +63,49 @@ def call(
     main_ax.set_axis_off()
     fig.add_axes(main_ax)
 
-    main_ax.imshow(
-        xarray_obj[product_name],
-        norm=mpl_colors_info["norm"],
-        cmap=mpl_colors_info["cmap"],
-    )
-
-    success_outputs = []
-    for fname in output_fnames:
-        LOG.info("Plotting %s with plt", fname)
-        # This just handles cleaning up the axes, creating directories, etc
-        success_outputs += save_image(
-            fig,
-            fname,
-            is_final=False,
-            image_datetime=xarray_obj.start_datetime,
-            savefig_kwargs=savefig_kwargs,
+    if is_3d:
+        slices = [
+            xarray_obj[product_name].data[slice]
+            for slice in range(xarray_obj[product_name].data.shape[0])
+        ]
+    else:
+        slices = [xarray_obj[product_name].data]
+    for slice_idx, slice_data in enumerate(slices):
+        main_ax.clear()
+        # Needs to be flipped upside-down as incoming data is flipped if unprojected.
+        # This is the case for family
+        # unsectored_xarray_dict_to_algorithm_to_output_format, but I'm not sure if it
+        # applies to all other families that use this output formatter. We should look
+        # into this.
```

Comment:
We should discuss this before we merge this PR.

Reply:
Unfortunately, I don't think this always works and we may need to figure out a different solution. Either that or we will need to chase down the reason for this.

When I run one of the existing tests, the result is now flipped and incorrect. I ran `./ami.WV-Upper.unprojected_image.sh` which should produce the first image below and now produces the second image.

![old](https://github.com/user-attachments/assets/f18b54c3-17b3-4170-b120-0fe3562c6af0)

![new](https://github.com/user-attachments/assets/a3bb72f4-6b70-4855-947b-df549d113d59)

Reply:
You're right that this should probably not flip the data upside down. Output formatters should just output, and not manipulate the data. Let me readdress the reader in my OVERCAST package and see if I can flip the data upside down then (or in my algorithms), instead of at this step.

Reply:
Yeah, I can revert that change. The OVCST data is naturally going from south to north and I can fix that within my reader. We should be able to resolve this once I've got that pushed.

Reply:
Pushed those changes; Jeremy feel free to resolve this if you're good with them!